woocommerce/woocommerce
# Update in-app product card clicked Tracks event

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:


The `wcadmin_marketplace_product_card_clicked` tracks event in the in-app marketplace currently uses the product and group names as properties, but these values can change, and can also return translated values, which makes it difficult to filter them in Tracks.

We're adding the group id and product id values to these events, as those are always the same.


Closes https://github.com/Automattic/woocommerce.com/issues/21526



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out this branch
2. Go to `/plugins/woocommerce` and run `pnpm --filter='@woocommerce/plugin-woocommerce' build`
3. Visit the [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions).
4. Open JS console and run `localStorage.setItem( 'debug', 'wc-admin:*' );`. This makes tracks logs show up in the console. While in development, tracks events are not sent up to the server.
5. On the dev console, click on Preserve Logs, so that even if the page reload, the recorded console logs are preserved.
6. Click on a product card on the Discover tab (Cmd-Click if you want to ensure you stay on the page).
7. Check the logs for a new debug entry for `marketplace_product_card_clicked`
8. Confirm the tracks properties include the `group_id` field, and that the `product` field shows the WCCOM product id. The product title should appear in the `product_name` field.
![Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite](https://github.com/user-attachments/assets/ce3aca98-38b5-48b3-b439-a7ee5419b26d)

9. Switch to the Extensions or Themes tab. Again click a product card. Confirm the logged event includes the `product` field with the product id, but the `group` and `group_id` fields should not be included now.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Include group and product id properties in the `marketplace_product_card_clicked` Tracks event
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
